backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -127,6 +128,25 @@ public DatasetVersionServiceImpl(
     return accessibleDatasetVersionIds;
   }
 
+  public List<String> getAccessibleDatasetVersionIDsByAction(
+      List<String> requestedDatasetVersionIds, ModelDBServiceActions modelDBServiceActions) {
+    // Validate if current user has access to the entity or not
+    if (requestedDatasetVersionIds.size() == 1) {
+      roleService.isSelfAllowed(
+          ModelDBServiceResourceTypes.DATASET_VERSION,
+          modelDBServiceActions,
+          requestedDatasetVersionIds.get(0));
+      return requestedDatasetVersionIds;
+    } else {
+      List<String> allowedDatasetVersionIds =
+          roleService.getSelfAllowedResources(
+              ModelDBServiceResourceTypes.DATASET_VERSION, modelDBServiceActions);
+      // Validate if current user has access to the entity or not
+      allowedDatasetVersionIds.retainAll(requestedDatasetVersionIds);
+      return allowedDatasetVersionIds;
```

Comment:
based on the new role definitions at https://github.com/VertaAI/verta-uacservice/pull/207 you need not make second level binding calls on get. 

You need to only check the binding for delete